analyses/tmb-vaf-longitudinal/04-tmb-unpaired-samples-plot.Rmd

```diff
@@ -0,0 +1,292 @@
+       path = plots_dir, 
+       width = 25, 
+       height = 20, 
+       device = "pdf", 
+       useDingbats = FALSE)
+```
+
+# Summary statistics for Timepoints and cancer types
+Considering the inequalities in the sampling effort for each timepoint, let's explore whether the sampling size affects the observed patterns, e.g., indication of Progressive and Recurrence having higer TMB compared to Diagnosis for High-grade glioma patients.
+
+```{r summary-statistics, fig.width = 8, fig.height = 6, fig.fullwidth = TRUE}
+# Inspect some random rows of the data by groups
+set.seed(2023)
+
+df_plot %>% 
+  select(cgGFAC, timepoint_cgGFAC_n, log10_tmb) %>% 
+  sample_n_by(cgGFAC, timepoint_cgGFAC_n, log10_tmb, size = 1) 
+
+# Define cancer_groups
+cancer_groups <- unique(as.character(df_plot$cgGFAC))
+cancer_groups <- sort(cancer_groups, decreasing = FALSE)
+print(cancer_groups)
+
+
+for (i in seq_along(cancer_groups)) {
+  print(i)
+  
+  df_sub <- df_plot %>% 
+    filter(cgGFAC == cancer_groups[i]) %>% 
+    arrange(tumor_descriptor, timepoint_cgGFAC_n) 
+  
+  # To add cancer group in the stat.test table
+  df_sub_list <- df_sub %>% 
+    select(cgGFAC, timepoint_cgGFAC_n) %>% 
+    unique()
+  
+  # Compute some summary statistics (mean and sd) by groups:
+  stat.summary <- df_sub %>%
+    group_by(cgGFAC, timepoint_cgGFAC_n) %>%
+    get_summary_stats(log10_tmb, type = "median_iqr") %>%
+    as.data.frame() 
+
+  stat.test <- df_sub %>% 
+    rstatix::wilcox_test(log10_tmb ~ timepoint_cgGFAC_n) %>%
+    add_significance() %>%
+    as.data.frame() %>% 
+    mutate(timepoint_cgGFAC_n = group1) %>% 
+    left_join(df_sub_list, by = c("timepoint_cgGFAC_n")) %>% 
+    select(-timepoint_cgGFAC_n)
+
+  # Save results for all 
+  if(i == 1){
+    stat.summary.all <- stat.summary
+    stat.test.all <- stat.test
+    } else {
+      stat.summary.all <- rbind(stat.summary.all, stat.summary)
+      stat.test.all <- rbind(stat.test.all, stat.test)
+    }
+    
+  
+  # Add add_xy_position
+  stat.test <- stat.test %>% 
+    add_xy_position(x = "timepoint_cgGFAC_n") # , step.increase = .08
+
+  # Create bxp
+  print(ggboxplot(df_sub,
+                  x = "timepoint_cgGFAC_n",
+                  y = "log10_tmb",
+                  color = "tumor_descriptor",
+                  palette = palette,
+                  add = c("mean")) + # add = c("mean", "jitter")
+          theme_Publication() + 
+          #rotate() +
+          theme(axis.text.x = element_text(angle = 90)) +
+          stat_pvalue_manual(stat.test,
+                             #tip.length = 0.01,
+                             hide.ns = TRUE, 
+                             label = "p.adj.signif",
+                             y.position = "y.position") +
+          scale_y_continuous(limits = c(0, max(stat.test$y.position))) +
+          labs(title = cancer_groups[i],
+               x = "Tumor descriptor",
+               y = "log10TMB",
+               color = "Timepoint"))
```

Comment:
I think there is an issue with the significance values in the plot not matching those in the results table. If you look at the plot for Other tumor there are significant TMB differences reported between Progressive and other time point, but the stats indicate these are actually differences between Diagnosis and other timepoints: 

```
         .y.              group1                   group2  n1  n2 statistic        p    p.adj p.adj.signif cgGFAC y.position       groups xmin xmax
1  log10_tmb     Deceased (N=25)        Diagnosis (N=920)  25 920    8887.0 5.20e-02 2.60e-01           ns  Other    1.63988 Deceased....    1    2
2  log10_tmb     Deceased (N=25)      Progressive (N=122)  25 122    1257.5 1.68e-01 5.04e-01           ns  Other    1.73308 Deceased....    1    3
3  log10_tmb     Deceased (N=25)       Recurrence (N=113)  25 113    1356.5 7.59e-01 8.18e-01           ns  Other    1.82628 Deceased....    1    4
4  log10_tmb     Deceased (N=25) Second Malignancy (N=13)  25  13     221.0 7.40e-02 2.97e-01           ns  Other    1.91948 Deceased....    1    5
5  log10_tmb   Diagnosis (N=920)      Progressive (N=122) 920 122   58699.0 4.09e-01 8.18e-01           ns  Other    2.01268 Diagnosi....    2    3
6  log10_tmb   Diagnosis (N=920)       Recurrence (N=113) 920 113   65360.0 7.67e-06 7.67e-05         ****  Other    2.10588 Diagnosi....    2    4
7  log10_tmb   Diagnosis (N=920) Second Malignancy (N=13) 920  13    9349.5 4.73e-04 4.00e-03           **  Other    2.19908 Diagnosi....    2    5
8  log10_tmb Progressive (N=122)       Recurrence (N=113) 122 113    8221.0 1.10e-02 7.50e-02           ns  Other    2.29228 Progress....    3    4
9  log10_tmb Progressive (N=122) Second Malignancy (N=13) 122  13    1200.5 2.00e-03 1.90e-02            *  Other    2.38548 Progress....    3    5
10 log10_tmb  Recurrence (N=113) Second Malignancy (N=13) 113  13    1031.0 1.80e-02 1.06e-01           ns  Other    2.47868 Recurren....    4    5
```
I am not 100% sure, but this may be caused by different leveling of factors in `stat.test` vs. `df_sub`. in df_sub (and plot) it is Diagnosis, Progressive, Recurrence, Deceased, Second Malignancy, but in stat.test it is alphabetical (so Deceased first).

Reply:
Thanks for catching this! `df_sub` was arranged by `tumor_descriptor` and `timepoint_cgGFAC_n`. See following code:

```
df_sub <- df_plot %>% 
    filter(cgGFAC == cancer_groups[i]) %>% 
    arrange(tumor_descriptor, timepoint_cgGFAC_n) 
```


I adjusted `stat.test` to match this but it didn't work (see 616fec8e2ea7443d65148226f7f0047d9257a41e). It turned out to be a bug related to the `add_xy_position()` function that misplaces the brackets on the ggplot (reported here https://github.com/kassambara/ggpubr/issues/477). I replaced it with `add_y_position()` (see 35e39dba3d131c89d3513e7e7abcf22c23f8d5e0) and the problem was fixed. :)  @rjcorb